lib/validator/v3/validator.go

```diff
@@ -813,6 +777,70 @@ func validateNetworkPolicy(v *validator.Validate, structLevel *validator.StructL
 	validateObjectMetaLabels(v, structLevel, np.Labels)
 }
 
+func validateGlobalNetworkPolicy(v *validator.Validate, structLevel *validator.StructLevel) {
+	gnp := structLevel.CurrentStruct.Interface().(api.GlobalNetworkPolicy)
+	spec := gnp.Spec
+
+	// Check the name is within the max length.
+	if len(gnp.Name) > k8svalidation.DNS1123SubdomainMaxLength {
+		structLevel.ReportError(
+			reflect.ValueOf(gnp.Name),
+			"Metadata.Name",
+			"",
+			reason(fmt.Sprintf("name is too long by %d bytes", len(gnp.Name)-k8svalidation.DNS1123SubdomainMaxLength)),
+		)
+	}
+
+	// Uses the k8s DN1123 label format for policy names.
+	matched := globalNetworkPolicyNameRegex.MatchString(gnp.Name)
+	if !matched {
+		structLevel.ReportError(
+			reflect.ValueOf(gnp.Name),
+			"Metadata.Name",
+			"",
+			reason("name must consist of lower case alphanumeric characters or '-' (regex: "+nameLabelFmt+")"),
+		)
+	}
+
+	validateObjectMetaAnnotations(v, structLevel, gnp.Annotations)
+	validateObjectMetaLabels(v, structLevel, gnp.Labels)
+
+	if spec.DoNotTrack && spec.PreDNAT {
+		structLevel.ReportError(reflect.ValueOf(spec.PreDNAT),
+			"PolicySpec.PreDNAT", "", reason("PreDNAT and DoNotTrack cannot both be true, for a given PolicySpec"))
+	}
+
+	if spec.PreDNAT && len(spec.Egress) > 0 {
+		structLevel.ReportError(reflect.ValueOf(spec.Egress),
+			"PolicySpec.Egress", "", reason("PreDNAT PolicySpec cannot have any Egress"))
```

Comment:
I appreciate you didn't change this, but I'd add "rules" at the end here.  (i.e. "... any Egress rules")